File: spec/python/test_process_xor_value.py
import unittest

from process_xor_value import ProcessXorValue

class TestProcessXorValue(unittest.TestCase):
    def test_process_xor_value(self):
        with ProcessXorValue.from_file("src/process_xor_1.bin") as r:

            self.assertEqual(r.key, 0xff)
            self.assertEqual(r.buf, b"foo bar")

File: spec/lua/test_process_xor_value.lua
local luaunit = require("luaunit")

require("process_xor_value")

TestProcessXorValue = {}

function TestProcessXorValue:test_process_xor_value()
    local r = ProcessXorValue:from_file("src/process_xor_1.bin")

    luaunit.assertEquals(r.key, 0xff)
    luaunit.assertEquals(r.buf, "foo bar")
end
